### DISCUSSION TITLE
How to get container content size? [iOS]

### DISCUSSION THREAD

**Discussion by wgywgy** [reactwg/react-native-new-architecture#82] [Q&A]
I want get the container contentsize, but In the new arch(Enable fabric), RCTRootView's method: - (void)rootViewDidChangeIntrinsicSize:(RCTRootView *)rootView;
isn't call. Which replace method can at right time get IntrinsicSize?
Mannal get RCTFabricSurface's intrinsicSize also get zero value.

RN Version
0.68.1

**Comment by wgywgy** [comment]
I need help, Thank you.

**Comment by wgywgy** [comment]
Fristly, I calc contentsize on RCTRootComponentView's method - (void)mountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index and - (void)unmountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index. (Union the componentView bounces). 
But sometime it's wrong.(In the case ScrollView)
Now I use js side onLayout event.nativeEvent.layout.width and height, and set native side contianer size by bridge method.
What's the recommand method?

**Comment by troZee** [comment]
hey @wgywgy 
Could you provide screenshots from layout [inspector](https://fbflipper.com/docs/features/plugins/inspector/) and point, which component is sized incorrectly ? 

Regarding you question: `How to get container content size? [iOS] `
Here you have an answer: https://github.com/facebook/react-native/blob/main/ReactCommon/react/renderer/components/scrollview/ScrollViewShadowNode.cpp#L30

**Reply by wgywgy** [reply]
which component is sized incorrectly ?
It's not sized incorrectly, my problem is I want know the container size.
The old arch tell by method - (void)rootViewDidChangeIntrinsicSize:(RCTRootView *)rootView.
The new arch does not call.

**Comment by hoanghuy-ng** [comment]
@wgywgy hi, i have same the issues above. Do you have a solution to solve this problem yet?
